Language/Classes/Constructors/Factories/syntax_t06.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A factory is a constructor prefaced by the built-in identifier
+/// factory.
+/// ⟨factoryConstructorSignature⟩ ::=
+///     const? factory ⟨constructorName⟩ ⟨formalParameterList⟩
+///
+/// @description Checks that it is a compile-time error if a non-redirecting
+/// factory constructor has an initializers list
+/// @author msyabro
+
+class A {
+  int x;
+  A(this.x);
+  factory A.f() : x = 1 {
+//        ^
+// [analyzer] unspecified
+// [cfe] unspecified
+    return C();
+  }
+}
+
+class C extends A {
+  C() : super(0);
+}
+
+enum E {
+  e1(1), e2(2);
+
+  final x;
+
+  const E(this.x);
+//             ^
+// [analyzer] unspecified
```

Comment:
Why would we expect an error here?

Reply:
I just tried to run this program in DartPad, and I can see the error message. However, it looks like a spurious error message (it says that `x` is initialized at the declaration, but that isn't true), so maybe the analyzer should be adjusted such that it does not emit that error message.

I created https://github.com/dart-lang/sdk/issues/54335 to suggest this change to the analyzer.

I think we should then land this test as it is, and adjust it at some point in the future when the error messages changes, if they do indeed make those changes to the analyzer.